test/loopback.test.js

```diff
@@ -739,4 +739,157 @@ describe('loopback', function() {
       });
     }
   });
+
+  describe('Hiding shared methods', function() {
+    var app;
+
+    beforeEach(setupLoopback);
+
+    it('hides remote methods using fixed method names', function() {
+      var TestModel = app.registry.createModel(uniqueModelName);
+      app.model(TestModel, {
+        dataSource: null,
+        methods: {
+          staticMethod: {
+            isStatic: true,
+            http: {path: '/static'},
+          },
+        },
+        options: {
+          remoting: {
+            sharedMethods: {
+              staticMethod: false,
+            },
+          },
+        },
+      });
+
+      var publicMethods = TestModel.sharedClass
+        .methods()
+        .filter(function(m) {
+          return m.shared === true;
+        })
+        .map(function(m) {
+          return m.stringName.replace(/^[^.]+\./, ''); // drop the class name
+        });
```

Comment:
Please extract these 8 lines of code into a shared helper function. E.g. `listPublicMethods(TestModel)`.